components/dashboards/org.wso2.carbon.dashboards.core/src/main/java/org/wso2/carbon/dashboards/core/internal/WidgetMetadataProviderImpl.java

```diff
@@ -68,10 +68,12 @@ protected void activate(BundleContext bundleContext) {
         try {
             if (dataSourceService != null && configProvider != null) {
                 this.widgetMetadataDao = WidgetMetadataDaoFactory.createDao(dataSourceService, configProvider);
+                this.widgetMetadataDao.initWidgetTable();
                 this.isDaoInitialized = true;
             }
         } catch (DashboardException e) {
             //ignore as its required to start with default widget extension loading.
+            LOGGER.debug("Error in activating widget DAO {}", e.getMessage());
```

Comment:
Seems like the exception `e` is swallowed, any reason? IMO, either we log the exception properly or wrap & re-throw.